flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendMigrationTestBase.java

```diff
@@ -1244,8 +1244,8 @@ void testStateMigrationAfterChangingTTLFromDisablingToEnabling() {
                                 testKeyedValueStateUpgrade(
                                         initialAccessDescriptor, newAccessDescriptorAfterRestore))
                 .satisfiesAnyOf(
-                        e -> assertThat(e).isInstanceOf(IllegalStateException.class),
-                        e -> assertThat(e).hasCauseInstanceOf(IllegalStateException.class));
+                        e -> assertThat(e).isInstanceOf(StateMigrationException.class),
+                        e -> assertThat(e).hasCauseInstanceOf(StateMigrationException.class));
     }
```

Comment:
Would you mind adding one test to check all state backends call the new `resolveSchemaCompatibility(TypeSerializerSnapshot)` instead of old method?

Reply:
As discussed in [#21635(comment)](https://github.com/apache/flink/pull/21635#discussion_r1426110489), I'd like to also add more state level test after adjusting in a seprate pr. WDYT?